platforms/software/maven/src/main/java/org/gradle/api/publish/maven/tasks/GenerateMavenPom.java

```diff
@@ -45,112 +45,57 @@
 @UntrackedTask(because = "Gradle doesn't understand the data structures used to configure this task")
 public abstract class GenerateMavenPom extends DefaultTask {
 
-    private final Transient.Var<MavenPom> pom = varOf();
-    private Object destination;
-    private final Cached<MavenPomFileGenerator.MavenPomSpec> mavenPomSpec = Cached.of(() ->
-        MavenPomFileGenerator.generateSpec((MavenPomInternal) getPom())
-    );
-
-    @Inject
-    protected FileResolver getFileResolver() {
-        throw new UnsupportedOperationException();
-    }
-
-    /**
-     * Get the version range mapper.
-     *
-     * @deprecated This method will be removed in Gradle 9.0
-     */
-    @Inject
-    @Deprecated
-    protected VersionRangeMapper getVersionRangeMapper() {
-        throw new UnsupportedOperationException();
-    }
-
-    /**
-     * The values set by this method are ignored.
-     *
-     * @deprecated This method will be removed in Gradle 9.0.
-     */
-    @Deprecated
-    public GenerateMavenPom withCompileScopeAttributes(ImmutableAttributes compileScopeAttributes) {
```

Comment:
I thought we agreed we should only include changes related to the provider api migration in this branch?

I understand these are trivially removed in 9.0, but removing these seems out of scope